pkg/management/delete_user_defined_alert_rule_by_id.go

```
@@ -119,14 +133,42 @@ func (c *client) deleteUserAlertRuleById(ctx context.Context, namespace, name, a
 	}
-	return nil
+	return c.cleanupARCForDeletedRule(ctx, namespace, name, alertRuleId)
+}
+
+// cleanupARCForDeletedRule attempts to remove any associated ARC after a rule is deleted.
+// It determines the ARC namespace from the rule's namespace and silently skips if no ARC exists.
+func (c *client) cleanupARCForDeletedRule(ctx context.Context, namespace, name, alertRuleId string) error {
+	nn := types.NamespacedName{Namespace: namespace, Name: name}
+	arcNamespace, err := c.arcNamespaceForRule(nn)
+	if err != nil {
+		return nil
```

golangci-lint (2.12.2)

[error] error is not nil (line 150) but it returns nil

(nilerr)